woocommerce/woocommerce-gutenberg-products-block
# add E2E tests infrastructure

This PR introduces E2E infrastructure so we can start writing some tests in that area, this is still a work on progress, there are some things that need to be done that I'm not fully sure yet how to do.

Mainly what needs to be done is:
- [ ] integrate other plugins in the local env.
- [ ] see a bug of why I need to pass `--` for the command to work.

### How to test the changes in this Pull Request:

1. run `npm install`
2. e2e tests will use your existing `localhost:8889` if you have one. If you don't, running `npm run env:start` should do that for you.
3. run `npm run test:e2e --` not sure why not passing `--` result in a timeout.

You can visualize your tests by running `npm run test:e2e -- --puppeteer-interactive`.

If you're using your `localhost:8889` make sure login credentials are `admin:password`.

Those tests are based on WordPress core e2e tests and Gutenberg tests, I've added a Travis job but not sure if we should keep it for now, it should always pass as long as WordPress is working.
